stdlib/public/Concurrency/Task.cpp

```diff
@@ -234,7 +193,7 @@ void NullaryContinuationJob::process(Job *_job) {
   auto *task = job->Task;
   auto *continuation = job->Continuation;
 
-  _swift_task_dealloc_specific(task, job);
+  free(job);
```

Comment:
This allocation from the task's async stack was messing with the assumption we had that the dependency record is always the last thing on the task's async stack. 

Given that this is used for `Task.yield` which I expect to be used rarely, it seemed fine to switch this to using malloc instead.

Reply:
cc: @drexin

Reply:
Only spotted this today; I think it's "fine" 

...but worth taking note of, because I know there's a group of people doing horrible things with spinning on Task.yield in order to try to get "reproducible test execution" which is a horrible pattern we should not encourage, but in case such folks come back and notice the regression in sleep/yield, I made a note/radar for ourselfes here: rdar://105718307